torchmultimodal/modules/losses/albef.py

```diff
@@ -127,106 +127,56 @@ class MaskedLanguageModelingLoss(nn.Module):
     Support loss distillation for non-zero alpha. Compute standard mlm loss for zero alpha.
 
     Args:
-        masked_token_id (int): The token id indicating a masked token. Default is -100.
-        vocab_size (int): The number of different tokens the prediction_head can predict. Default is 30522.
-        hidden_size (int): The hidden size of the prediction_head. Default is 768.
-        layer_norm_eps (float): The epsilon used by the prediction_head normalization layer. Default is 1e-12.
-        alpha (float): The interpolation value between mlm_loss and loss_distill. Default is 0.
-        transform_act_fn (Callable[[Tensor], Tensor]): The activation function in the prediction_head. Default is GELU.
+        mask_token_id (int): The token id indicating a masked token. Default is -100.
 
     Inputs:
         labels (Tensor of shape (batch_size, seq_length)): The masked output tokens.
-        hidden_states (Tensor of shape (batch_size, seq_length, hidden_size)):
-            The hidden states of preceding tokens.
-        hidden_states_m (Optional[Tensor] of shape (batch_size, seq_length, hidden_size)):
-            The hidden states of preceding tokens from momentum models.
+        prediction_scores (Tensor of shape (batch_size, seq_length, vocab_size)):
+            The prediction scores from a prediction head.
+        prediction_scores_m (Optional[Tensor] of shape (batch_size, seq_length, vocab_size)):
+            The prediction scores from a momentum prediction head.
             Required if alpha is non-zero.
+        alpha (float): The interpolation value between mlm_loss and loss_distill. Default is 0.
     """
 
     def __init__(
         self,
         mask_token_id: int = -100,
-        vocab_size: int = 30522,
-        hidden_size: int = 768,
-        layer_norm_eps: float = 1e-12,
-        alpha: float = 0.0,
-        transform_act_fn: Callable[[Tensor], Tensor] = nn.functional.gelu,
     ) -> None:
         super().__init__()
         self.mask_token_id = mask_token_id
-        self.vocab_size = vocab_size
-        self.alpha = alpha
-        self.prediction_head = PredictionHead(
-            vocab_size, hidden_size, layer_norm_eps, transform_act_fn
-        )
 
     def forward(
         self,
         labels: Tensor,
-        hidden_states: Tensor,
-        hidden_states_m: Optional[Tensor] = None,
+        prediction_scores: Tensor,
+        prediction_scores_m: Optional[Tensor] = None,
+        alpha: Optional[float] = 0.0,
```

Comment:
what was the reason for moving alpha outside of the `__init__`? seems more suitable for an attribute

Reply:
The training loop passes in different `alpha` values during warm up so I thought it makes sense for it to be part of the func call. Another option is to keep it in `__init__` and change `model.loss.alpha` in the training loop. Which one is more preferable?

Reply:
ok that makes sense, let's keep it in forward